doc/specs/#2697 - PowerShell DSC resources.md

```diff
@@ -0,0 +1,256 @@
+---
+author: Ruben Guerrero @msftrubengu
+created on: 2022-11-14
+last updated: 2022-11-14
+issue id: 2697
+---
+
+# PowerShell DSC Resources for WinGet Spec
+
+For [#2697](https://github.com/microsoft/winget-cli/issues/2697)
+
+## Abstract
+
+This document contains the initial proposal for WinGet PowerShell DSC resources that help configure WinGet to the users desired state.
+
+## Inspiration
+
+See referenced [#2697](https://github.com/microsoft/winget-cli/issues/2697).
+
+## Solution Design
+
+A new module Microsoft.WinGet.DSC is going to be created where all the WinGet related Powershell DSC resources are going to live. The resources are simple wrappers around functionality that Microsoft.WinGet.Client currently provides, with some additions. We are going to use PSDesiredStateConfiguration 3.0.0 for development, so the new module is going to initially be an alpha release.
+
+The proposed DSC resources are (names TBD):
+1. WinGetUserSettings
+2. WinGetAdminSettings
+3. WinGetSources
+4. WinGetIntegrity
```

Comment:
To me it feels better to have one resource per package, but I get that we want to have one that takes the output of `winget export`. Maybe we can have both? Implementation-wise it will be mostly similar except parsing the json/hashtable.